### PR TITLE
feat: use processor to display remark reports directly

### PR DESCRIPTION
**Pull request by JounQin**
This will make the report much more familiar with original `remark-lint` plugins.

image

### CODE REVIEW

**Comment by JounQin** [comment]
@wooorm I'm still confused how to set report severity in remark config, or should I add a setting in `eslint-plugin-mdx` instead.

File: packages/eslint-plugin-mdx/src/processors/index.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { remark } from './remark'
+
+export const processors = { remark }
```
Comment (JounQin) [inline packages/eslint-plugin-mdx/src/processors/index.ts:3]: related https://github.com/benmosher/eslint-plugin-import/issues/2002

File: packages/eslint-plugin-mdx/src/rules/remark.ts [hunk 2]
```diff
@@ -59,15 +58,20 @@ export const remark: Rule.RuleModule = {
...
+              fatal ? 2 : fatal == null ? 0 : 1,
+          }
           context.report({
-            messageId: 'remarkReport',
-            data: {
-              reason,
-              source,
-              ruleId,
-            },
+            message: JSON.stringify(message),
```
Comment (JounQin) [inline packages/eslint-plugin-mdx/src/rules/remark.ts:74]: related https://github.com/eslint/eslint/issues/14198

**Comment by wooorm** [comment]
> I'm still confused how to set report severity in remark config, or should I add a setting in eslint-plugin-mdx instead.

Severity cannot be configured in a reporter, because it has no clue how serious things are.
Plugins default to warning. They can sometimes be configured.

**Comment by JounQin** [comment]
> They can sometimes be configured.

`sometimes` seems bad, is there any plan to improve it like eslint? I saw may requests about it.

**Comment by wooorm** [comment]
> is there any plan to improve it like eslint? I saw may requests about it.

What did you see?
How could it be improved?

**Comment by JounQin** [comment]
> What did you see?
> How could it be improved?

https://github.com/remarkjs/remark-lint/issues?q=+severity

**Comment by JounQin** [comment]
https://github.com/remarkjs/remark-lint/issues/65#issuecomment-240178240

**Comment by wooorm** [comment]
🤷‍♂️

These are issues that include the word severity. They do not show what you mean...

**Comment by JounQin** [comment]
> 🤷‍♂️
> 
> These are issues that include the word severity. They do not show what you mean...

@wooorm Check https://github.com/remarkjs/remark-lint/issues/65#issuecomment-240178240 instead.

**Comment by wooorm** [comment]
That’s my comment. From 5 years ago. The issue is closed because severities were added back then. 5 years ago...

> It seems not all rules support this option.

Please back this up. What are you running into? What is the problem?

**Comment by wooorm** [comment]
To clarify: all remark lint rules allow this.

But unified !== remark-lint.

ESLint is only a linter. unified is 500 projects. Some of them don’t support configuring severities.

**Comment by JounQin** [comment]
Sorry, my problem seems related to https://github.com/remarkjs/remark-lint/issues/98

So `["lint-no-duplicate-headings",[2]]` is working actually.🤣

**Comment by JounQin** [comment]
@wooorm Then this PR is ready to review and merge now!

**Comment by JounQin** [comment]
@mdx-js/releasers @wooorm Please help to release a minor version.

**Comment by wooorm** [comment]
done

**Pull request merged**